### ISSUE TITLE
[bug] csv_provider Docker image fails to start

### ISSUE THREAD

**Issue by sophokles73** [eclipse/kuksa.val.feeders#119]
With 665440f008c44bfa7363e36cca8ad4969bb7f3a4 the Docker image created for the csv_provider no longer starts up but fails with the following error:
```sh
> docker run ghcr.io/eclipse/kuksa.val.feeders/csv-provider:main
Traceback (most recent call last):
  File "provider.py", line 128, in <module>
  File "asyncio/runners.py", line 44, in run
  File "asyncio/base_events.py", line 649, in run_until_complete
  File "provider.py", line 79, in main
  File "kuksa_client/grpc/aio.py", line 55, in __init__
TypeError: BaseVSSClient.__init__() got an unexpected keyword argument 'tls_server_name'
[7] Failed to execute script 'provider' due to unhandled exception!
```

My understanding is that the BaseVSSClient's [*init* function](https://github.com/eclipse/kuksa.val/blob/9a7934fb8ae791f94049567b6ec2f2d12c5ab8b4/kuksa-client/kuksa_client/grpc/__init__.py#L451) does not accept arbitrary kwargs at that position in *kuksa-client* 0.3.x.

The init function's signature seems to have been changed in 0.4.0a3 to support this. However, the `pip install` command used in the csv_provider's Dockerfile does not match against pre-release versions and therefore picks up the incompatible kuksa-client 0.3.1 instead.

I see two ways forward here:
1. adjust the Dockerfile's `pip install` command to include `--pre`
2. do not include the *tls_server_name* argument when constructing the [VSSClient](https://github.com/eclipse/kuksa.val.feeders/blob/333e0c96dc27265554ffe51086da5918f0d471bd/csv_provider/provider.py#L80)

FMPOV option 1 would be more appropriate and also seems to have been anticipated by #93.
I would be happy to provide a PR ...

@erikbosch WDYT?

**Comment by erikbosch** [comment]
I think alternative 1 is the best. I have also written down as we as part of release should remove any `--pre`, see https://github.com/eclipse/kuksa.val.feeders/wiki/Release-Process

But we might have another problem, in [requirements.txt](https://github.com/eclipse/kuksa.val.feeders/blob/main/csv_provider/requirements.txt) we state that version shall be >0.3. Looking at the build log it seems to use a cached version. The intention of `>0.3` is that it shall complain/fail if it cannot find any 0.4+ version, but that does not seem to happen. So maybe we need to change the build config as well. I did a quick test at https://github.com/eclipse/kuksa.val.feeders/pull/120 but it is stuck in the queue, will check tomorrow.

**Comment by erikbosch** [comment]
I think this one can be closed as #120 is merged
